packages/sitecore-jss-nextjs/src/editing/editing-config-middleware.test.ts

```diff
@@ -0,0 +1,120 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { spy } from 'sinon';
+import { expect } from 'chai';
+import { EditingConfigMiddleware } from './editing-config-middleware';
+import { QUERY_PARAM_EDITING_SECRET } from './constants';
+
+type Query = {
+  [key: string]: string;
+};
+
+const mockRequest = (method: string, query?: Query) => {
+  return {
+    method,
+    query: query ?? {},
+  } as NextApiRequest;
+};
+
+const mockResponse = () => {
+  const res = {} as NextApiResponse;
+  res.status = spy(() => {
+    return res;
+  });
+  res.json = spy(() => {
+    return res;
+  });
+  res.end = spy(() => {
+    return res;
+  });
+  return res;
+};
+
+const componentsArray = ['TestComponentOne', 'TestComponentTwo'];
+const componentsMap = new Map<string, unknown>();
+componentsMap.set('TestComponentOne', {});
+componentsMap.set('TestComponentTwo', {});
+const metadata = { packages: { testPackageOne: '0.1.1' } };
+
+const expectedResult = {
+  components: ['TestComponentOne', 'TestComponentTwo'],
+  packages: { testPackageOne: '0.1.1' },
+};
+
+describe('EditingConfigMiddleware', () => {
+  const secret = 'jss-editing-secret-mock';
+
+  beforeEach(() => {
+    process.env.JSS_EDITING_SECRET = secret;
+  });
+
+  after(() => {
+    delete process.env.JSS_EDITING_SECRET;
+  });
+
+  it('should respond with 401 for missing secret', async () => {
```

Comment:
Should we also add a test case in case of an error or we don't need error handling here other than the secret?

Reply:
I think it should be fine. there is not much more logic there and if something fails it should be caught on component builder/metadata side